jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,36 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+#define JERRY_CONTEXT(field) (((jerry_context_t *) jerry_port_get_current_instance()->context_p)->field)
```

Comment:
Yes, less indirection would help, that is why we have several macros rather than a single one (which would be enough theoretically, but in practice it is not efficient).

As for me I would prefer some kind of strong inlining here, that was the purpose of the macro originally. When you write:

    JERRY_CONTEXT (a) = 5;
    JERRY_CONTEXT (b) = 5;
    JERRY_CONTEXT (c) = 5;
    JERRY_CONTEXT (d) = 5;

The compiler should be able to recognize that the base is the same. Somehow tell the compiler that the base is non-volatile.